nexus-sled-agent-shared/src/inventory.rs

```diff
@@ -591,6 +593,15 @@ impl ZoneKind {
     }
 }
 
+#[derive(
+    Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash,
+)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum OmicronZoneImageSource {
+    InstallDataset,
```

Comment:
I think it's worth some detailed comments here.  Here's a suggestion:

```suggestion
    /// Specifies that this zone's image should come from whatever happens to be on the sled's "install" dataset
    ///
    /// This is whatever was put in place in the factory or the latest MUPdate.  The image used here can vary by sled and even over time (if the sled gets MUPdated).
    /// Historically, this was the only source for zone images.  Once we deliver automated, control-plane-driven update, we expect to only use this value in emergencies where the control plane was offline and could not be recovered except via MUPdate.  
    InstallDataset,
```